### ISSUE TITLE
[BUG]: macos-12/13 microsoft hosted agent keeps hanging. Intermittently on net7 and constantly on net8

### ISSUE THREAD

**Issue by takla21** [microsoft/azure-pipelines-agent#4677]
### What happened?

The `macos12` & `macos13` hosted agent seems to hang intermittently when building our `net7-ios` project. When migrating to `net8.0-ios`, it always hangs.

![instability1](https://github.com/microsoft/azure-pipelines-agent/assets/17461593/caf83814-5c12-405d-82dc-9b10fe9c1d0e)
![instability2](https://github.com/microsoft/azure-pipelines-agent/assets/17461593/527df4e7-cbb1-4a27-8692-7b8b4c447a04)
![instability3](https://github.com/microsoft/azure-pipelines-agent/assets/17461593/0b3f4959-cf83-47af-ae48-f483b10b2bc5)
![instability4](https://github.com/microsoft/azure-pipelines-agent/assets/17461593/e5c020b5-4efd-4f66-b988-e0117f2cc789)

Here's the task that hangs
```yml
- bash: |
    cd ${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile
    dotnet publish -f:net7.0-ios -c:Release /p:ApplicationFlavor=$(ApplicationFlavor) /p:ApplicationIdentifier=$(ApplicationIdentifier) /p:CodesignProvision=$(provisioningProfile.provisioningProfileUuid) /p:ArchiveOnBuild=true /bl:$(build.ArtifactStagingDirectory)/build-net7.0-ios.binlog "/p:InformationalVersion=$GITVERSION_INFORMATIONALVERSION" /p:IsLightBuild=false /p:FirebaseCrashlyticsUploadSymbolsEnabled=false
  displayName: Build project for Release - $(BuildTargetFramework)
```

Here's the content of the mobile project building
```csproj
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<LangVersion>10.0</LangVersion>
		<TargetFrameworks>net7.0-android;net7.0-ios</TargetFrameworks>
		<SingleProject>true</SingleProject>
		<GenerateDocumentationFile>true</GenerateDocumentationFile>
		<OutputType>Exe</OutputType>
		<IsUnoHead>true</IsUnoHead>
		<UseMauiEssentials>true</UseMauiEssentials>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net7.0-android'">29.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net7.0-ios'">14.2</SupportedOSPlatformVersion>
		<NoWarn>Uno0001</NoWarn>
		<FirebaseCrashlyticsUploadSymbolsEnabled>True</FirebaseCrashlyticsUploadSymbolsEnabled>
	</PropertyGroup>
	<!-- Linker configuration. -->
	<PropertyGroup>
		<PublishTrimmed>True</PublishTrimmed>
		<TrimMode>partial</TrimMode>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="Chinook.BackButtonManager.Uno.WinUI" Version="1.1.2" />
		<PackageReference Include="Chinook.DataLoader.Uno.WinUI" Version="1.2.0" />
		<PackageReference Include="Chinook.DynamicMvvm.Uno.WinUI" Version="1.1.2" />
		<PackageReference Include="Chinook.SectionsNavigation.Uno.WinUI" Version="1.1.2" />
		<PackageReference Include="ExtendedSplashScreen.Uno.WinUI" Version="1.0.2" />
		<PackageReference Include="MessageDialogService.Uno.WinUI" Version="0.6.0-dev.58" />
		<PackageReference Include="Nventive.Persistence.Uno.WinUI" Version="0.4.0-dev.36" />
		<PackageReference Include="Nventive.View.Uno.WinUI" Version="0.5.0-dev.70" />
		<PackageReference Include="Reactive.Annex.Uno.WinUI" Version="0.6.0-dev.50" />
		<PackageReference Include="ReviewService.NativePrompters" Version="1.0.0" />
		<PackageReference Include="Uno.CommunityToolkit.WinUI" Version="7.1.100" />
		<PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" />
		<PackageReference Include="Uno.Material.WinUI" Version="3.0.40" />
		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Interactivity.WinUI" Version="2.3.1-uno.2" />
		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.3.1-uno.2" />
		<PackageReference Include="Uno.Toolkit.WinUI.Material" Version="4.2.22" />
		<PackageReference Include="Uno.WinUI" Version="4.10.34" />
		<PackageReference Include="Uno.WinUI.RemoteControl" Version="4.10.34" Condition="'$(Configuration)'=='Debug'" />
		<PackageReference Include="Uno.WinUI.Lottie" Version="4.10.34" />
		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.10.34" />
		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
		<PackageReference Include="Serilog.Sinks.Xamarin" Version="1.0.0" />
		<PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
		<PackageReference Include="Xamarin.Build.Download" Version="0.11.4" />
		<PackageReference Include="OneSignalSDK.DotNet" Version="4.3.3" />
                <!-- And many more internal packages that targets net6.0, net6.0-android, net6.0-ios-->
	</ItemGroup>
	<ItemGroup>
		<ProjectReference Include="..\ProjectName.Access\ProjectName.Access.csproj" />
		<ProjectReference Include="..\ProjectName.Business\ProjectName.Business.csproj" />
		<ProjectReference Include="..\ProjectName.Presentation\ProjectName.Presentation.csproj" />
	</ItemGroup>
	<!-- Disable Bugsee for Production. -->
	<ItemGroup Condition="'$(ApplicationEnvironment)'!='Production'">
		<PackageReference Include="Bugsee" Version="3.0.0" />
	</ItemGroup>
	<ItemGroup>
		<Folder Include="Android\Resources\color\" />
	</ItemGroup>
	<Choose>
		<When Condition="'$(TargetFramework)'=='net7.0-android'">
			<PropertyGroup>
				<AndroidCreatePackagePerAbi>False</AndroidCreatePackagePerAbi>
				<BundleAssemblies>False</BundleAssemblies>
			</PropertyGroup>

			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<PropertyGroup>
						<AndroidUseSharedRuntime>True</AndroidUseSharedRuntime>
						<UseHighDPIResources>False</UseHighDPIResources>
						<EmbedAssembliesIntoApk>False</EmbedAssembliesIntoApk>
						<EnableLLVM>False</EnableLLVM>
						<AndroidEnableProfiledAot>False</AndroidEnableProfiledAot>
						<AndroidDexTool>d8</AndroidDexTool>
						<Optimize>False</Optimize>
						<RunAOTCompilation>False</RunAOTCompilation>
					</PropertyGroup>
				</When>
				<When Condition="'$(Configuration)'=='Release'">
					<PropertyGroup>
						<EmbedAssembliesIntoApk>True</EmbedAssembliesIntoApk>
						<AndroidManagedSymbols>True</AndroidManagedSymbols>
						<AndroidEnableSGenConcurrent>True</AndroidEnableSGenConcurrent>
						<AndroidAotAdditionalArguments>nodebug,no-write-symbols</AndroidAotAdditionalArguments>
						<AndroidUseDefaultAotProfile>False</AndroidUseDefaultAotProfile>
						<AndroidUseAapt2>True</AndroidUseAapt2>
					</PropertyGroup>
					<Choose>
						<When Condition="'$(IsLightBuild)'=='True'">
							<!-- Build Optimization for LightBuild (Faster Builds in Pull Requests). -->
							<PropertyGroup>
								<AndroidEnableProfiledAot>False</AndroidEnableProfiledAot>
								<EnableLLVM>False</EnableLLVM>
								<Optimize>False</Optimize>
								<RunAOTCompilation>False</RunAOTCompilation>
							</PropertyGroup>
						</When>
						<Otherwise>
							<PropertyGroup>
								<AndroidEnableProfiledAot>True</AndroidEnableProfiledAot>
								<!-- LLVM currently doesn't work with AOT enabled and custom profile. -->
								<!-- See https://github.com/dotnet/runtime/issues/74021 for more details. -->
								<!-- See https://github.com/xamarin/xamarin-android/issues/7352 for more details. -->
								<EnableLLVM>False</EnableLLVM>
								<Optimize>True</Optimize>
								<RunAOTCompilation>True</RunAOTCompilation>
							</PropertyGroup>
						</Otherwise>
					</Choose>
				</When>
			</Choose>
			<ItemGroup>
                                <!-- And many more internal packages that targets net6.0-android -->
				<PackageReference Include="Uno.UniversalImageLoader" Version="1.9.36" />
				<PackageReference Include="Mono.AotProfiler.Android" Version="7.0.0" />
				<PackageReference Include="Xamarin.Kotlin.StdLib.Jdk8" Version="1.9.20" />
				<PackageReference Include="Xamarin.Firebase.Crashlytics" Version="118.4.3.1" />
				<PackageReference Include="Xamarin.Firebase.Analytics" Version="121.3.0.3" />
				<PackageReference Include="Xamarin.Firebase.Perf " Version="120.5.0" />
				<PackageReference Include="Xamarin.GooglePlayServices.Wallet" Version="119.2.1.1" />
				<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData" Version="2.6.1.2" />
				<PackageReference Include="Xamarin.GooglePlayServices.Location" Version="121.0.1.3" />

				<!-- Those two packages are mandatory, otherwise, we can't compile with the latest AndroidX packages. See https://github.com/xamarin/AndroidX/issues/764 for more details. -->
				<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.3.0.1" />
				<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.0.1" />
			</ItemGroup>
			<ItemGroup>
				<AndroidAotProfile Include="custom.aprof" />
				<AndroidEnvironment Include="Android\environment.conf" />
				<TrimmerRootDescriptor Include="Android\LinkerExclusions.xml" />
			</ItemGroup>

			<!-- Android resources. -->
			<!-- Shared resources. -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\values\Colors.xml" />
				<AndroidResource Include="Android\Resources\values\Styles.xml" />
				<AndroidResource Include="Android\Resources\drawable\splash.xml" />
				<AndroidResource Include="Android\Resources\xml\file_paths.xml" />
			</ItemGroup>

			<!-- ApplicationFlavor specific resources. -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\drawable\*.*">
					<Link>Android\Resources\drawable\%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\drawable-*dpi\*.*">
					<Link>Android\Resources\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\mipmap-*dpi\*.*">
					<Link>Android\Resources\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\values\Strings.xml">
					<Link>Android\Resources\values\Strings.xml</Link>
				</AndroidResource>
			</ItemGroup>

			<!-- Network security configuration (For Fiddler). -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\xml\network_security_config.xml" />
			</ItemGroup>
			<!-- Android Firebase configuration. -->
			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<ItemGroup>
						<GoogleServicesJson Include="Android\google-services.json">
							<Link>google-services.json</Link>
						</GoogleServicesJson>
					</ItemGroup>
				</When>
				<Otherwise>
					<ItemGroup>
						<GoogleServicesJson Include="Android\Resources_$(ApplicationFlavor)\google-services.json">
							<Link>google-services.json</Link>
						</GoogleServicesJson>
					</ItemGroup>
				</Otherwise>
			</Choose>
			<!-- Google api key -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\values\maps.xml" />
			</ItemGroup>
		</When>
		<When Condition="'$(TargetFramework)'=='net7.0-ios'">
			<PropertyGroup>
				<MtouchExtraArgs>$(MtouchExtraArgs) --setenv=MONO_GC_PARAMS=soft-heap-limit=512m,nursery-size=64m,evacuation-threshold=66,major=marksweep,concurrent-sweep</MtouchExtraArgs>
				<MtouchExtraArgs>$(MtouchExtraArgs) --xml=./iOS/LinkerExclusions.xml --linkskip=$(AssemblyName)</MtouchExtraArgs>
				<!-- See https://github.com/unoplatform/uno/issues/9430 for more details. -->
				<MtouchExtraArgs>$(MtouchExtraArgs) --registrar:static</MtouchExtraArgs>
				<!-- See https://github.com/xamarin/xamarin-macios/issues/14812 for more details. -->
				<MtouchExtraArgs>$(MtouchExtraArgs) --marshal-objectivec-exceptions:disable</MtouchExtraArgs>
				<CodesignEntitlements>iOS\Entitlements.plist</CodesignEntitlements>
				<!-- Workaround for clang++ error on iOS simulator on ARM base macOS system. See "https://github.com/dotnet/maui/issues/16778#issuecomment-1683343255" for more detail. -->
				<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
			</PropertyGroup>
			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<PropertyGroup>
						<MtouchDebug>True</MtouchDebug>
						<MtouchExtraArgs>$(MtouchExtraArgs) --setenv=MONO_LOG_LEVEL=debug --setenv=MONO_LOG_MASK=gc</MtouchExtraArgs>
						<Optimize>False</Optimize>
						<ProvisioningType>manual</ProvisioningType>
					</PropertyGroup>
					<PropertyGroup Condition="'$(Platform)'=='iPhone'">
						<CodesignKey>iPhone Developer</CodesignKey>
					</PropertyGroup>
				</When>
				<When Condition="'$(Configuration)'=='Release'">
					<PropertyGroup>
						<BuildIpa>True</BuildIpa>
						<CodesignKey>iPhone Distribution</CodesignKey>
						<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
					</PropertyGroup>
					<Choose>
						<When Condition="'$(IsLightBuild)'=='True'">
							<!-- Build Optimization for LightBuild (Faster Builds in Pull Requests). -->
							<PropertyGroup>
								<Optimize>False</Optimize>
								<!-- Prevents PNG Optimization. -->
								<OptimizePNGs>False</OptimizePNGs>
								<!-- Disables LLVM. -->
								<MtouchUseLlvm>False</MtouchUseLlvm>
								<MtouchUseThumb>False</MtouchUseThumb>
								<!-- Disables Symbol Stripping. -->
								<MtouchNoSymbolStrip>True</MtouchNoSymbolStrip>
								<!-- Disables DSYM Creation. -->
								<MtouchExtraArgs>$(MtouchExtraArgs) --dsym=false</MtouchExtraArgs>
							</PropertyGroup>
						</When>
						<Otherwise>
							<PropertyGroup>
								<OptimizePNGs>True</OptimizePNGs>
								<MtouchUseLlvm>True</MtouchUseLlvm>
								<MtouchUseThumb>True</MtouchUseThumb>
								<MtouchNoSymbolStrip>False</MtouchNoSymbolStrip>
								<Optimize>True</Optimize>
							</PropertyGroup>
						</Otherwise>
					</Choose>
				</When>
			</Choose>
			<ItemGroup>
				<PackageReference Include="Uno.Extensions.Logging.OSLog" Version="1.4.0" />
				<PackageReference Include="Xamarin.Firebase.iOS.Crashlytics" Version="8.10.0.3" />
				<PackageReference Include="Xamarin.Firebase.iOS.Analytics" Version="8.10.0.3" />
				<PackageReference Include="Xamarin.Firebase.iOS.PerformanceMonitoring" Version="8.10.0.3" />

				<!-- The latest version of SkiaSharp is required with Firebase. See https://github.com/unoplatform/uno/discussions/14009 for more details -->
				<PackageReference Include="SkiaSharp" Version="2.88.3" />
			</ItemGroup>
			<ItemGroup>
				<TrimmerRootDescriptor Include="iOS\LinkerExclusions.xml" />
			</ItemGroup>

			<!-- iOS resources -->
			<!-- Shared resources. -->
			<ItemGroup>
				<!--
				Removing this fix release build.
				<InterfaceDefinition Include="iOS\LaunchScreen.storyboard" />
				-->
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet.png" />
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet%403x.png" />
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet%402x.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet.png.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet%402x.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet%403x.png" />
			</ItemGroup>
			<!-- ApplicationFlavor specific resources. -->
			<ItemGroup>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\Assets\**\*.*">
					<Link>iOS\Resources\Assets\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</BundleResource>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\en.lproj\**\*.*">
					<Link>iOS\Resources\en.lproj\%(RecursiveDir)%(FileName)%(Extension) </Link>
				</BundleResource>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\fr.lproj\**\*.*">
					<Link>iOS\Resources\fr.lproj\%(RecursiveDir)%(FileName)%(Extension) </Link>
				</BundleResource>
				<ImageAsset Include="iOS\Resources_$(ApplicationFlavor)\Media.xcassets\**\*.*">
					<Link>Resources\Media.xcassets\%(RecursiveDir)%(FileName)%(Extension)</Link>
					<Visible>false</Visible>
				</ImageAsset>
			</ItemGroup>
			<!-- iOS Firebase configuration. -->
			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<ItemGroup>
						<!-- Needed to upload dSYM file in debug mode and not get warning email. -->
						<BundleResource Include="iOS\GoogleService-Info.plist">
							<Link>GoogleService-Info.plist</Link>
						</BundleResource>
					</ItemGroup>
				</When>
				<Otherwise>
					<ItemGroup>
						<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\GoogleService-Info.plist">
							<Link>GoogleService-Info.plist</Link>
						</BundleResource>
					</ItemGroup>
				</Otherwise>
			</Choose>
		</When>
	</Choose>
	<Import Project="..\ProjectName.Shared.Views\ProjectName.Shared.Views.projitems" Label="Shared" />

	<!--Workaround for issue https://github.com/dotnet/runtime/issues/90032-->
	<ItemGroup Condition="'$(Configuration)'=='Release'">
		<TrimmableAssembly Include="CommunityToolkit.WinUI.UI" />
	</ItemGroup>

	<!-- Workaround to remove bitcode introduced by OneSignal. See https://github.com/OneSignal/OneSignal-DotNet-SDK/issues/27 for more details. -->
	<Import Project="..\..\..\BitcodeRemovalWorkaround.targets" Condition="'$(TargetFramework)' == 'net7.0-ios'" />
</Project>
```

### Versions

Agent version: 3.234.0
macos-12 specs: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

macOS 12.7.3 

### Relevant log output

It keeps hanging around that spot

```shell
  ProjectName.Mobile -> /Users/runner/work/1/s/src/app/ProjectName.Mobile/bin/Release/net7.0-ios/ios-arm64/ProjectName.Mobile.dll
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/parasol.activities/0.40.5/lib/net6.0-ios16.1/Parasol.Activities.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.fonts.fluent/2.3.0/lib/netstandard1.0/Uno.Fonts.Fluent.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.fonts.roboto/2.2.2/lib/netstandard2.0/Uno.Fonts.Roboto.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.toolkit.winui/4.2.22/lib/net6.0-ios15.4/Uno.Toolkit.WinUI.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.toolkit.winui.material/4.2.22/lib/net6.0-ios15.4/Uno.Toolkit.WinUI.Material.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Access/bin/Release/net7.0/ProjectName.Access.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Business/bin/Release/net7.0/ProjectName.Business.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Presentation/bin/Release/net7.0/ProjectName.Presentation.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file 'bin/Release/net7.0-ios/ios-arm64/ProjectName.Mobile.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
  /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip
  /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
ILLINK : warning MT0079: The recommended Xcode version for Microsoft.iOS 16.4.7099 is Xcode 14.3 or later. The current Xcode version (found in /Applications/Xcode_14.2.app/Contents/Developer) is 14.2. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
  		
./iOS/LinkerExclusions.xml(9,3): warning IL2007: Could not resolve assembly 'ProjectName.DataAccess'. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
iOS/LinkerExclusions.xml(9,3): warning IL2007: Could not resolve assembly 'ProjectName.DataAccess'. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework/GoogleUtilitiesComponents : warning MT7091: The framework /Users/runner/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/FAnlytcs-8.9.1/FirebaseAnalytics-8.9.1/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_armv7/FirebaseAnalytics.framework/FirebaseAnalytics : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/FAnlytcs-8.9.1/FirebaseAnalytics-8.9.1/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_armv7/FirebaseAnalytics.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GAppM-8.9.1/GoogleAppMeasurement-8.9.1/Frameworks/GoogleAppMeasurement.xcframework/ios-arm64_armv7/GoogleAppMeasurement.framework/GoogleAppMeasurement : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GAppM-8.9.1/GoogleAppMeasurement-8.9.1/Frameworks/GoogleAppMeasurement.xcframework/ios-arm64_armv7/GoogleAppMeasurement.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Base/Frameworks/GoogleMapsBase.framework/GoogleMapsBase : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Base/Frameworks/GoogleMapsBase.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMapsCore.framework/GoogleMapsCore : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMapsCore.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMaps.framework/GoogleMaps : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMaps.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
##[error]The Operation will be canceled. The next steps may not contain expected logs.
##[error]The operation was canceled.
Finishing: Build project for Release - net7.0-ios
```

**Comment by DenisNikulin5** [comment]
Hi @takla21. Thank you for your question!

Do you have the ability to check if it works fine on a self-hosted agent?

**Comment by takla21** [comment]
After further testing on `net8.0-ios` it always hangs compared to `net7.0-ios`. I've updated the issue title & description

**Comment by takla21** [comment]
I was able to find a workaround for the hanging issues by setting `MtouchUseLlvm` to false. I've opened an issue on xamarin-macios https://github.com/xamarin/xamarin-macios/issues/20210. Maybe it's an issue outside of the hosted agent

**Comment by DenisNikulin5** [comment]
@takla21 Thank you for your investigation!
Based on the information you provided, this isn't the agent's fault.
Please let us know if you'll have any updates, otherwise, I guess it makes sense to close the issue.

**Comment by mikklfr** [comment]
Hello, we had similar issues on Mac M2 Pro self hosted agents with randomly looping/hanging jobs regardless of the current tasks.
It happened on many different tasks from flutter compilation, bash script execution to push of artifacts onto azure devops.

We found it was because of multi concurrency issues and scaled down to only one agent per machine without success.

We found out that our machines had the x86 version of the vsts agent and some tasks were actually using rosetta to perform.
Moving from x86 vsts to arm vsts based agents solved all of our issues.
